server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -303,7 +302,8 @@ public void merge(IndexMetaData indexMetaData, MergeReason reason) {
     }
 
     public DocumentMapper merge(String type, CompressedXContent mappingSource, MergeReason reason) {
```

Comment:
It would be good to remove `MapperService#getTypeForUpdate`, since it is no longer used.